api/handlers.go

```diff
@@ -60,32 +81,28 @@ func (api *API) userHandlerPOST(w http.ResponseWriter, req *http.Request, ps htt
 }
 
 // userHandlerPUT updates an existing user.
-func (api *API) userHandlerPUT(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
-	err := req.ParseMultipartForm(MaxMultipartMem)
+func (api *API) userHandlerPUT(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	uid, _, _, err := jwtToken(req)
 	if err != nil {
-		WriteError(w, errors.New("Failed to parse multipart parameters."), http.StatusBadRequest)
+		WriteError(w, err, http.StatusInternalServerError)
 		return
 	}
-	var u *user.User
-	// Fetch the user by their _id.
-	if id := req.PostFormValue("_id"); id != "" {
-		u, err = api.staticDB.UserByID(req.Context(), id)
-		if err != nil {
-			// This is a Bad Request and not an Internal Server Error because
-			// the client has supplied an invalid `_id`.
-			WriteError(w, errors.AddContext(err, "failed to fetch user"), http.StatusBadRequest)
-			return
-		}
-	}
-	// Fetch the user by their email.
-	if u == nil {
-		email := (user.Email)(req.PostFormValue("email"))
-		u, err = api.staticDB.UserByEmail(req.Context(), email)
-		if err != nil {
-			WriteError(w, errors.AddContext(err, "failed to fetch user"), http.StatusBadRequest)
-			return
-		}
+
+	var u *database.User
+	// Fetch the user by their id.
+	u, err = api.staticDB.UserByID(req.Context(), uid)
+	if err != nil {
+		// This is a Bad Request and not an Internal Server Error because
+		// the client has supplied an invalid user id.
+		WriteError(w, errors.AddContext(err, "failed to fetch user"), http.StatusBadRequest)
+		return
+	}
+	err = req.ParseMultipartForm(MaxMultipartMem)
```

Comment:
Why do we need this? `PostFormValue` should implicitly call that iff `r.PostForm == nil`.
It will also call it with about half the memory of `MaxMultipartMem` which seems reasonable.

Reply:
Oh, you're right. I think I wasn't getting my parameters for some reason and I read that this is what I needed to do. And then it worked, so I kept it. Probably it was something completely unrelated like a bad Postman setup.

Removed.